chunk.go

```diff
@@ -0,0 +1,43 @@
+package typhon
+
+import (
+	"io"
+	"net/http"
+)
+
+func copyChunked(dst io.Writer, src io.Reader) (written int64, err error) {
+	flusher, flusherOk := dst.(http.Flusher)
+	if !flusherOk {
+		return io.Copy(dst, src)
+	}
+
+	// This is taken and lightly adapted from the source of io.Copy
+	buf := make([]byte, 32*1024)
+	for {
+		nr, er := src.Read(buf)
+		if nr > 0 {
+			nw, ew := dst.Write(buf[0:nr])
+			if nw > 0 {
+				written += int64(nw)
+			}
+			if ew != nil {
+				err = ew
+				break
+			}
+			if flusherOk {
+				flusher.Flush()
+			}
+			if nr != nw {
+				err = io.ErrShortWrite
+				break
+			}
+		}
+		if er != nil {
```

Comment:
Almost caught me here with `er`, but all good 😄

Reply:
Yeah, I copied and pasted this directly from the source of the `io` package as the comment above mentions. I considered changing this too but thought it would probably be easier to maintain if I changed as little as possible. 🤔